tests/common/plugins/sanity_check/checks.py

```diff
@@ -100,41 +100,58 @@ def check_interfaces(dut):
 def check_bgp_status(dut):
 
     def _check_bgp_status_helper():
-        neighbors.clear()
-        neighbors.update(dut.get_bgp_neighbors())
-        if neighbors:
-            down_neighbors = [k for k, v in neighbors.items()
-                              if v['state'] != 'established']
-            if down_neighbors:
-                check_result['down_neighbors'] = down_neighbors
-                check_result['failed'] = True
+        asic_check_results = []
+        bgp_facts = dut.bgp_facts(asic_index='all')
+        for asic_index, a_asic_facts in enumerate(bgp_facts):
+            a_asic_result = False
+            a_asic_neighbors = a_asic_facts['ansible_facts']['bgp_neighbors']
+            if a_asic_neighbors:
+                down_neighbors = [k for k, v in a_asic_neighbors.items()
+                                  if v['state'] != 'established']
+                if down_neighbors:
+                    if dut.facts['num_asic'] == 1:
+                        check_result['bgp'] = {'down_neighbors' : down_neighbors }
+                    else:
+                        check_result['bgp' + str(asic_index)] = {'down_neighbors' : down_neighbors }
```

Comment:
Can you elaborate when we need to differentiate between single and multi asic here ? 
Ideally we should try to  avoid having if/else for num_asics as much as possible.

Reply:
In check_results (for which we do a json dump as a log message), we want to show which bgp instance has down_neighbors. So, for single asic, we want to show 'bgp', while for multi-asic, we want to show 'bgp0' or 'bgp1' etc.

For example, failure with single asic
```
        {
            "check_item": "bgp", 
            "failed": true, 
            "bgp": {
                "down_neighbors": [
                    "2064:101::1", 
                    "100.0.0.1", 
                    "101.0.0.1", 
                    "2064:100::1"
                ]
            }
        }, 
```

But for multi-asic, where we have down_neighbors only on bgp0, and not bgp1 and bgp2

```
        {
            "check_item": "bgp", 
            "failed": true, 
            "bgp0": {
                "down_neighbors": [
                    "100.0.0.1"
                ]
            }
        }, 
```

The other way to solve this would be that dut.bgp_facts(asic_index='all') in MultiAsicSonicHost returns a dictionary with key being the asic instance (like asic0, asic1 or global), rather than a list of results for each asic.